### PR TITLE
chore: update to support pandoc 3.3

### LINKED ISSUE #446: Support Pandoc 3.3
- https://github.com/Homebrew/homebrew-core/pull/178812

cc @lierdakil

### PR DESCRIPTION
**Pull request by chenrui333**
fixes #446

- https://github.com/Homebrew/homebrew-core/pull/178812

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -14,241 +14,3 @@ ghc-options:
...
-- ansi-terminal-1.1.1
-- ansi-terminal-types-1.1
-- appar-0.1.8
-- array-0.5.6.0
-- asn1-encoding-0.9.6
-- asn1-parse-0.9.5
-- asn1-types-0.3.4
-- assoc-1.1.1
-- async-2.2.5
-- attoparsec-0.14.4
```
Comment (chenrui333) [inline stack.yaml:38]: not quite sure why these got removed.

Comment (lierdakil) [reply stack.yaml:38]: Something got borked :shrug: Anyway, thanks, I'll have a look.

Comment (lierdakil) [reply stack.yaml:38]: So... apparently either you don't have `sed`, it doesn't behave like the makefile expects it to, or `cabal.project.freeze` was somehow missing. No idea tbh. "it works fine on my machine" :sweat_smile:

**Comment by lierdakil** [comment]
Merged via ed8704b31d6aae648163e0cd713b14fe86ed7c89.

Had to amend a bit, for one because of stack, for two because apparently your cabal uses GHC 9.8, and I'm not ready to upgrade the compiler on CI at the moment. Maybe I should pin the compiler in `cabal.project`, but that's going to mess with anyone downstream who builds from repo, probably.

I'll cut a new binary release once (if) CI finishes, so probably in a few hours.

**Comment by chenrui333** [comment]
no worries, thanks for the help!

**Pull request closed without merging**